bot/seasons/evergreen/TriviaQuiz.py

```diff
@@ -0,0 +1,263 @@
+        if ctx.channel.id in list(self.games.keys()):
+            return await ctx.send("Game already running in this channel!")
+
+        if category is None:
+            category = random.choice(list(self.categories.keys()))
+
+        else:
+            category = category.lower()
+            if category not in self.categories.keys():
+                embed = self.category_embed()
+                return await ctx.send(f"Category {category} does not exist!", embed=embed)
+
+        self.games[ctx.channel.id] = GameData(
+            owner=ctx.author.id,
+            category=category
+        )
+
+        await self.send_question(ctx.channel)
+
+    def category_embed(self):
+        """A function which returns an embed showing all avilable categories"""
+        embed = discord.Embed(colour=discord.Colour.blue())
+        embed.title = "The available question categories are:"
+        embed.description = ""
+        for cat, description in self.categories.items():
+            embed.description += f"**- {cat.capitalize()}**\n{description.capitalize()}\n"
+        embed.set_footer(text="If not category is chosen, then a random one will be selected.")
+        return embed
+
+    async def send_question(self, channel):
+        """This function is to be called whenever a question needs to be sent."""
+        await asyncio.sleep(2)
+        game = self.games[channel.id]
+        if game.unanswered_questions == self.inactivity_limit:
+            del self.games[channel.id]
+            return await channel.send("Game stopped due to inactivity.")
+
+        category = game.category
+        category_dict = self.questions[category]
+        question_dict = random.choice(category_dict)
+
+        same_question = True
+
+        # Making sure a question is not repeated.
+        while same_question is True:
+            question_dict = random.choice(category_dict)
+            if question_dict["id"] not in game.done_questions:
+                same_question = False
+            else:
+                pass
+
+        # Initial points for a question.
+        question_dict["points"] = 100
+        game.question = question_dict
+        game.hints = 0
+
+        embed = discord.Embed(colour=discord.Colour.dark_gold())
+
+        question = question_dict["question"]
+        question_id = question_dict["id"]
+
+        game.done_questions.append(question_id)
+        question_number = len(game.done_questions)
+
+        embed.title = f"#{question_number} Question"
+        embed.description = question
+        embed.set_footer(text="A hint will be provided after every 10s if no one gives the right answer.Max of 2 hints")
+
+        await channel.send(embed=embed)
+        await self.send_hint(channel, question_dict)
+
+    @commands.Cog.listener()
+    async def on_message(self, message):
+        """A function triggered when a message is sent."""
+        channel = message.channel
+        if channel.id not in list(self.games.keys()):
```

Comment:
There shouldn't be a need to create a list out of dictionary keys.
```suggestion
        if channel.id not in self.games:
```